src/components/CheckBoxFilterContainer.ts

```diff
@@ -0,0 +1,203 @@
+import { Component, ReactElement, createElement } from "react";
+import { findDOMNode } from "react-dom";
+
+import { Alert } from "./Alert";
+import { CheckboxFilter, CheckboxFilterProps } from "./CheckBoxFilter";
+import { Utils, parseStyle } from "../utils/ContainerUtils";
+import { DataSourceHelper } from "../utils/DataSourceHelper/DataSourceHelper";
+
+import * as classNames from "classnames";
+import * as dijitRegistry from "dijit/registry";
+import * as dojoConnect from "dojo/_base/connect";
+
+interface WrapperProps {
+    class: string;
+    style: string;
+    friendlyId: string;
+    mxform: mxui.lib.form._FormBase;
+    mxObject: mendix.lib.MxObject;
+}
+
+export interface ContainerProps extends WrapperProps {
+    listViewEntity: string;
+    filterBy: FilterOptions;
+    attribute: string;
+    attributeValue: string;
+    constraint: string;
+    defaultChecked: boolean;
+    unCheckedFilterBy: FilterOptions;
+    unCheckedAttribute: string;
+    unCheckedAttributeValue: string;
+    unCheckedConstraint: string;
+}
+
+type FilterOptions = "attribute" | "XPath" | "None";
+interface HybridConstraint {
+    attribute: string;
+    operator: string;
+    value: string;
+    path?: string;
+}
+
+export interface ListView extends mxui.widget._WidgetBase {
+    _datasource: {
+        _constraints: HybridConstraint[] | string;
+        _pageObjs: mendix.lib.MxObject[];
+    };
+    datasource: {
+        type: "microflow" | "entityPath" | "database" | "xpath";
+    };
+    __customWidgetDataSourceHelper: DataSourceHelper;
```

Comment:
Should be optional
`__customWidgetDataSourceHelper?: DataSourceHelper;`

Reply:
Done